src/pixelator/pixeldataset/__init__.py

```diff
@@ -0,0 +1,400 @@
+                             defaults to None
+        :param colocalization: a `pd.DataFrame` with colocalization information,
+                               defaults to None
+        :param copy: specify if the input data should be copied or not.
+                     Defaults to True.
+        :param allow_empty_edgelist: allow the edgelist to be empty. Defaults to False.
+        :return: An instance of PixelDataset
+        :rtype: PixelDataset
+        """
+        return PixelDataset(
+            backend=ObjectBasedPixelDatasetBackend(
+                adata=adata,
+                edgelist=edgelist,
+                metadata=metadata,
+                polarization=polarization,
+                colocalization=colocalization,
+                copy=copy,
+                allow_edgelist_to_be_empty=allow_empty_edgelist,
+            )
+        )
+
+    @property
+    def adata(self) -> AnnData:
+        """Get the AnnData object."""
+        return self._backend.adata
+
+    @adata.setter
+    def adata(self, value: AnnData) -> None:
+        """Set the AnnData object."""
+        self._backend.adata = value
+
+    @property
+    def edgelist(self) -> pd.DataFrame:
+        """Get the edge list."""
+        return _enforce_edgelist_types(self._backend.edgelist)
+
+    @edgelist.setter
+    def edgelist(self, value: pd.DataFrame) -> None:
+        """Set the edge list."""
+        self._backend.edgelist = _enforce_edgelist_types(value)
+
+    @property
+    def edgelist_lazy(self) -> pl.LazyFrame:
+        """Get the edge list as a lazy dataframe."""
+        lz_edgelist = self._backend.edgelist_lazy
+        if "index" in lz_edgelist.columns:
+            warnings.warn(
+                "A column called `index` was identified in your edgelist. "
+                "This will be removed."
+            )
+            lz_edgelist = lz_edgelist.drop("index")
+        return lz_edgelist
+
+    @property
+    def polarization(self) -> Optional[pd.DataFrame]:
+        """Get the polarization object."""
+        return self._backend.polarization
+
+    @polarization.setter
+    def polarization(self, value: pd.DataFrame) -> None:
+        """Set the polarization object."""
+        self._backend.polarization = value
+
+    @property
+    def colocalization(self) -> Optional[pd.DataFrame]:
+        """Get the colocalization object."""
+        return self._backend.colocalization
+
+    @colocalization.setter
+    def colocalization(self, value: pd.DataFrame) -> None:
+        """Set the colocalization object."""
+        self._backend.colocalization = value
+
+    @property
+    def metadata(self) -> Dict:
+        """Get the metadata dictionary."""
+        metadata = self._backend.metadata
+        if not metadata:
+            return {}
+        return metadata
+
+    @metadata.setter
+    def metadata(self, value: Dict) -> None:
+        """Set the metadata object."""
+        self._backend.metadata = value
```

Comment:
It seems clear we are not testing the setters.

Reply:
Yeah. However, I suggest we stick to only the refactoring in this PR, and go on with improving test coverage later. I think it's easier when large scale moves (as this one) are not conflated with adding/removing code.